### ISSUE TITLE
Remove Block keyboard shortcut causes display server on Linux to shut down or reboot

### ISSUE THREAD

**Issue by ZebulanStanphill** [WordPress/gutenberg#9036]
## The issue
Currently, the Remove Block keyboard shortcut is <kbd><kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>Backspace</kbd></kbd>. This works fine on Windows and macOS. But on Linux, if you are using the X Window System or Weston (the reference Wayland implementation), this keyboard shortcut will cause the display server to shut down or restart. Obviously, this is quite a surprise to anyone who is just trying to delete a block, and can certainly cause some unexpected data loss.

Tested on Antergos Linux using X11 and KDE Plasma.

**Comment by designsimply** [comment]
I searched for documentation about keyboard shortcuts for Linux to see if I could find a good reference for `Ctrl+Alt+Backspace` and the closest I could find is `Ctrl+Alt+Delete` is for logging out in KDE/GNOME ([ref](https://en.wikipedia.org/wiki/Table_of_keyboard_shortcuts#Navigation)) and on some Linux-based operating systems including Ubuntu and Debian ([ref](https://en.wikipedia.org/wiki/Control-Alt-Delete#Linux)) and, on Ubuntu Server, it is used to reboot a computer without logging in.

**Comment by ZebulanStanphill** [comment]
@designsimply

https://utcc.utoronto.ca/~cks/space/blog/unix/XBackspaceTerminateHistory

https://www.howtogeek.com/howto/linux/disable-ctrlaltbackspace-from-restarting-x-windows-on-linux/

It is also mentioned in this table:
https://en.wikipedia.org/wiki/Control-Alt-Delete#Equivalents_on_various_platforms

It looks like the shortcut is disabled by default in some distros like Ubuntu. Apparently it is not in Antergos (and presumably also Arch Linux).

**Comment by aldavigdis** [comment]
This is a pretty serious issue. ctrl-alt-backspace or ctrl-alt-del is a common shortcut to restart the xorg server.

**Comment by talldan** [comment]
It's very very easy to change the shortcut, but very very hard to find combinations that work and don't conflict with something.

Would be good to have some suggestions of shortcuts that might work. I think just about every combination with backspace/delete/d has been ruled out now.

**Comment by jasmussen** [comment]
Could we use the minus key instead of the Delete key? Ctrl + Shift + Minus?

**Comment by ZebulanStanphill** [comment]
@jasmussen That does not seem to already be in use for anything on Antergos Linux using Firefox, so I think we could! :+1:

**Comment by talldan** [comment]
☝️ Made a PR to change it to Ctrl+Shift+-

Seems to work in my testing on Windows & Mac. Would be great if someone could test on Linux.

The only thing that looks a bit odd is the way the shortcut is displayed in the block settings menu (screenshot in the PR description). I think it's ok though, if tackled, we should do it on a separate issue.

**Comment by mcsf** [comment]
This is an interesting issue to bring forward. Indeed, `Ctrl+Alt+Backspace` is not ideal in Linux. There are ways around this, including remapping the shortcut only for Linux or for all platforms, but I just wanted to leave a drive-by thought:

Perhaps a root problem is that our [`primaryAlt`](https://github.com/WordPress/gutenberg/blob/6d1d284d900b71ba44c226ba6694c09200bca695/packages/keycodes/src/index.js#L50) modifier maps to `Cmd+Opt` in macOS and `Ctrl+Alt` elsewhere. I'd argue, though this is subjective, that — while `Cmd+Opt` is a common prefix for many application-level hotkeys in macOS tradition — `Ctrl+Alt` feel idiomatically more "serious", not uncommonly dealing with system-level interactions.

**Comment by afercia** [comment]
> It's very very easy to change the shortcut, but very very hard to find combinations that work and don't conflict with something.

It is indeed. I'd say it's a known issue since years and years, especially to people who specialize in keyboard accessibility. Operating systems, browsers, extensions, keyboard layouts, not to mention assistive technologies, etc. bring in a huge amount of keyboard shortcuts and it's basically impossible to guarantee there won't be conflicts. As I see it, the only solution in the long term would be considering a mechanism to allow users to remap shortcuts, as proposed in https://github.com/WordPress/gutenberg/issues/3218 (now moved to projects > ideas). I'd agree that's something to consider for a future version.

**Comment by talldan** [comment]
@afercia I have some ideas on how that could be achieved. It basically consists of moving the definition of shortcut keys into a very simple redux store. It sort of kills two birds with one stone - a central place to define keyboard shortcuts (at the moment they're all over the codebase) and also allowing the future possibility of remapping keys.

**Comment by afercia** [comment]
> a central place to define keyboard shortcuts (at the moment they're all over the codebase) 

@talldan that would be awesome 🎉(also for maintenance, handbook documentation)

**Comment by talldan** [comment]
Reopening since #9190 has been reverted.

**Comment by talldan** [comment]
Have opened a new pull request to address this - #10008

Would be great to get some testing, especially on linux. 😄

**Comment by lkraav** [comment]
Hi @talldan. I just got this unpleasant surprise today for the first time, as I was exploring upgrading my Gutenberg keyboard skill.

#10008 seems to have been merged, but I'm on 3.9.0 and X still got killed. Gutenblock context menu still says "Remove block Ctrl+Alt+Backspace". I'm I supposed to see or experience something different?

**Comment by talldan** [comment]
Hi @lkraav - the fix should be in 4.0.0. Sorry for stopping your display server 😭
